src/components/svgs/nodes/mrt-num-line-badge.tsx

```diff
@@ -0,0 +1,124 @@
+import { RmgFields, RmgFieldsField } from '@railmapgen/rmg-components';
+import { CityCode, MonoColour } from '@railmapgen/rmg-palette-resources';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { AttrsProps } from '../../../constants/constants';
+import { MiscNodeType, Node, NodeComponentProps } from '../../../constants/nodes';
+import { AttributesWithColor, ColorField } from '../../panels/details/color-field';
+
+const MRTNumLineBadge = (props: NodeComponentProps<MRTNumLineBadgeAttributes>) => {
+    const { id, x, y, attrs, handlePointerDown, handlePointerMove, handlePointerUp } = props;
+    const { num = defaultMRTNumLineBadgeAttributes.num, color = defaultMRTNumLineBadgeAttributes.color } =
+        attrs ?? defaultMRTNumLineBadgeAttributes;
+
+    const onPointerDown = React.useCallback(
+        (e: React.PointerEvent<SVGElement>) => handlePointerDown(id, e),
+        [id, handlePointerDown]
+    );
+    const onPointerMove = React.useCallback(
+        (e: React.PointerEvent<SVGElement>) => handlePointerMove(id, e),
+        [id, handlePointerMove]
+    );
+    const onPointerUp = React.useCallback(
+        (e: React.PointerEvent<SVGElement>) => handlePointerUp(id, e),
+        [id, handlePointerUp]
+    );
+
+    const fgColor = color[3];
+    const bgColor = color[2];
+
+    return React.useMemo(
+        () => (
+            <g
+                id={id}
+                transform={`translate(${x}, ${y})`}
+                onPointerDown={onPointerDown}
+                onPointerMove={onPointerMove}
+                onPointerUp={onPointerUp}
+                style={{ cursor: 'move' }}
+            >
+                <rect fill={bgColor} x="-6" y="-6" width="12" height="12" rx="6" ry="6" />
```

Comment:
Why just not use a `<circle />` element?

Reply:
I remembered that there would be different between safari and edge/chrome if using circle when I made chongqingRT num line badge.

Reply:
Hummmm, is it possible to upload some screenshots? Safari is a hard browser to set the environment up :(

Reply:
I changed it, and no error thrown this time.